internal/p2p/pqueue_test.go

```diff
@@ -0,0 +1,45 @@
+package p2p
+
+import (
+	"testing"
+	"time"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+func TestCloseWhileDequeueFull(t *testing.T) {
+	enqueueLength := 5
+	chDescs := []ChannelDescriptor{
+		{ID: 0x01, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x02, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x03, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x04, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x05, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x06, Priority: 1, MaxSendBytes: 4},
```

Comment:
```suggestion
		{ID: 0x01, Priority: 1, MaxSendBytes: 4},
```
nit: you could just have a single channel